### DISCUSSION TITLE
How to block Jint to call indexer to find `then` function

### DISCUSSION THREAD

**Discussion by viceice** [sebastienros/jint#2006] [Q&A] [answered]
We've some object which has a string indexer and throws on unknown keys. we like to return it from async contexts.

```cs
class Dummy {
 public object this[string key] => key == "then" ? throw new Exeption(): key;
}

var engine = new Engine(o =>
  {
    o.Strict().CatchClrExceptions();
    o.ExperimentalFeatures = ExperimentalFeature.All;
  }
);

engine.Global.FastSetDataProperty("di", JsValue.FromObject(engine, new Dummy()));
engine.Evaluate("(async function(){ return di; })()").UnwrapIfPromise(); // throws
```

Our Object is more complex, but it will throw for `then`. How can we suppress that it will try the indexer for `then`? It needs to be accessable for other purposes from js.

**Comment by lahma** [comment]
Maybe a custom object wrapper is required because someone might want to provide custom then function from c# type.

**Reply by viceice** [reply]
is there an easy way to extend the existing wrapper? the specific object has a lot a children with more functions added directly, which also needs to be callable.

**Reply by lahma** [reply, marked as answer]
```c#
var engine = new Engine(o => o.Interop.MemberAccessor = (_, _, member) => member == "then" ? JsValue.Undefined : null);
```
